### PR TITLE
Remove useless AppendVertices.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Issue(s) number: 
Close https://github.com/vesoft-inc/nebula/issues/4273

#### Description:
Remove the useless AppendVertices executor.

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by czpmango — DISMISSED** [review]

File: src/graph/optimizer/rule/EliminateAppendVerticesRule.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    if (expr != nullptr) {
+      return false;
+    }
+  }
+  const auto* appendVertices = static_cast<const AppendVertices*>(matched.planNode({0, 0}));
+  DCHECK_EQ(appendVertices->kind(), graph::PlanNode::Kind::kAppendVertices);
+  if (appendVertices->vFilter() != nullptr || appendVertices->filter() != nullptr) {
+    return false;
+  }
+  if (appendVertices->colNames().back()[0] != '_') {  // Anonymous node
```
Comment (CPWstatic) [inline src/graph/optimizer/rule/EliminateAppendVerticesRule.cpp:53]: We should have a function to do this, like `AnonVarGenerator::isAnno(string varName)`

File: src/graph/optimizer/rule/EliminateAppendVerticesRule.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+using nebula::graph::AppendVertices;
+using nebula::graph::PlanNode;
+using nebula::graph::Project;
+using nebula::graph::QueryContext;
+
+namespace nebula {
+namespace opt {
+
+// TODO If we prune properties in Optimizer Rules, we could check properties of AppendVertices,
+// and drop AppendVertices if it has no properties directly.
```
Comment (CPWstatic) [inline src/graph/optimizer/rule/EliminateAppendVerticesRule.cpp:22]: That's right, but more conditions should be considered, such as if the var is used by other operators.
```
Match (a)-[]-(b) Return id(b);
```

**Review by CPWstatic — COMMENTED** [review]
Excellent!

File: src/graph/optimizer/rule/EliminateAppendVerticesRule.cpp (hunk 2 above)
Comment (Shylock-Hg) [reply src/graph/optimizer/rule/EliminateAppendVerticesRule.cpp:22]: In this case, there is properties `_vid`, it's not empty too.

File: src/graph/optimizer/rule/EliminateAppendVerticesRule.cpp (hunk 1 above)
Comment (Shylock-Hg) [reply src/graph/optimizer/rule/EliminateAppendVerticesRule.cpp]: Done.

**Review by CPWstatic — APPROVED** [review]

**Review by czpmango — APPROVED** [review]

**Pull request merged**